### PR TITLE
Add support for long value return for CEIL, CEILING and FLOOR math functions

### PR DESCRIPTION
**Pull request by MitchellGale-BitQuill**
Add support for long value return for CEIL, CEILING and FLOOR math functions.

Ceil and ceiling should round up to the nearest whole number if there is a decimal value.
Floor should round down to the nearest whole number if there is a decimal value.


Signed-off-by: MitchellGale-BitQuill <mitchellg@bitquilltech.com>

### Description
Currently the max size returned by OpenSearchSQL for values called with `ceil`, `ceiling`, `floor` is `INT_MAX`, this PR adds additional support for some values up to `LONG` return value.

There are concerns regarding accepted values because of issues with floating point precision issues. Please see this comment for more detail https://github.com/opensearch-project/sql/issues/1036#issuecomment-1335790282.

Issue exists in the positive domain for `ceil` and `ceiling` as well as the negative domain for `floor`.

Example issue `ceil`:
```
opensearchsql> select ceil(282474973688888.022);                                                                                                                                                                                                            
fetched rows / total rows = 1/1
+-----------------------------+
| ceil(282474973688888.022)   |
|-----------------------------|
| 282474973688888             |
+-----------------------------+
opensearchsql> select ceil(282474973688888.122);                                                                                                                                                                                                            
fetched rows / total rows = 1/1
+-----------------------------+
| ceil(282474973688888.122)   |
|-----------------------------|
| 282474973688889             |
+-----------------------------+
```

Example issue `floor`:
```
opensearchsql> select floor(-282474973688888.122);                                                                                                                                 
fetched rows / total rows = 1/1
+-------------------------------+
| floor(-282474973688888.122)   |
|-------------------------------|
| -282474973688889              |
+-------------------------------+
opensearchsql> select floor(-282474973688888.022);                                                                                                                                 
fetched rows / total rows = 1/1
+-------------------------------+
| floor(-282474973688888.022)   |
|-------------------------------|
| -282474973688888              |
+-------------------------------+
```



### Notes
Because of the change of data type, several unit test, integration tests and comparison tests needed modification to reflect change from `INT` -> `LONG`.
 
### Issues Resolved
https://github.com/opensearch-project/sql/issues/1036
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docs/user/dql/functions.rst [hunk 1]
```diff
@@ -234,9 +234,63 @@ CEIL
 Description
 >>>>>>>>>>>
 
-Specifications:
+Usage: CEIL(a) takes the ceiling of value a.
+
+Limitation: CEIL only works as expected when IEEE 754 double type displays decimal when stored.
+
+Argument type: a: double
```
Comment (Yury-Fridlyand) [inline docs/user/dql/functions.rst:241]:
```suggestion
Argument type: <any numeric type>
```
I don't know how to write better - list all types or leave a placeholder. The function can round an integer too, result would be `LONG` anyway.
Please, update for all functions.

File: docs/user/dql/functions.rst [hunk 2]
```diff
@@ -234,9 +234,63 @@ CEIL
...
+    | 3147483648               | 113147483648               | 3147483648               |
+    +--------------------------+----------------------------+--------------------------+
+
+CEILING
+----
+
+Description
+>>>>>>>>>>>
+
+Usage: CEILING(a) takes the ceiling of value a.
```
Comment (Yury-Fridlyand) [inline docs/user/dql/functions.rst:269]: Can you add cross links between `CEIL` and `CEILING`?

File: docs/user/dql/functions.rst [hunk 3]
```diff
@@ -424,10 +478,37 @@ FLOOR
 Description
 >>>>>>>>>>>
 
-Specifications:
+Usage: FLOOR(a) takes the floor of value a.
 
-1. FLOOR(NUMBER T) -> T
+Argument type: a: double
```
Comment (Yury-Fridlyand) [inline docs/user/dql/functions.rst:483]: Same as above.
Could you add limitation too?

File: integ-test/src/test/java/org/opensearch/sql/correctness/runner/connection/JDBCConnection.java [hunk 4]
```diff
@@ -17,6 +17,7 @@
 import java.sql.Statement;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Objects;
```
Comment (Yury-Fridlyand) [inline integ-test/src/test/java/org/opensearch/sql/correctness/runner/connection/JDBCConnection.java:20]: a mistake?

**Review by Yury-Fridlyand — APPROVED** [review]

**Comment by Yury-Fridlyand** [comment]
Do you need to update PPL docs and tests too?

**Comment by GumpacG** [comment]
Maybe something to note in the PR description would be if this is a breaking change. For example, the taco needs to be tested against this change. There is a overwritten function that expects `CEIL` and `FLOOR` to return an `int`. 
https://github.com/opensearch-project/sql/blob/43ceda18930c459ff4f3c50651d98063f4bcc334/bi-connectors/TableauConnector/src/dialect.tdd#L305

**Comment by Yury-Fridlyand** [comment]
There is no `long` in tdd:
> Allowable types: none, bool, real, int, str, datetime, date, localstr, null, error, any, tuple, spatial, localreal, localint
[ref](https://tableau.github.io/connector-plugin-sdk/docs/dialect#:~:text=the%20correct%20order.-,Allowable%20types,-%3A%20none%2C%20bool%2C%20real)

**Comment by GumpacG** [comment]
> There is no `long` in tdd:
> 
> > Allowable types: none, bool, real, int, str, datetime, date, localstr, null, error, any, tuple, spatial, localreal, localint
> 
> [ref](https://tableau.github.io/connector-plugin-sdk/docs/dialect#:~:text=the%20correct%20order.-,Allowable%20types,-%3A%20none%2C%20bool%2C%20real)

Right. In which case the return type would be `real` which would potentially use the default generate formula by Tableau instead of the specified formula in the dialect.

File: docs/user/dql/functions.rst (hunk 1 above)
Comment (MitchellGale-BitQuill) [reply docs/user/dql/functions.rst]: Addressed in b5fd1208eb4d4012ce8a7fa16b93a2b50ac4eaf1

File: docs/user/dql/functions.rst (hunk 2 above)
Comment (MitchellGale-BitQuill) [reply docs/user/dql/functions.rst:271]: Addressed (I think) in b5fd1208eb4d4012ce8a7fa16b93a2b50ac4eaf1

File: docs/user/dql/functions.rst (hunk 3 above)
Comment (MitchellGale-BitQuill) [reply docs/user/dql/functions.rst]: Addressed in b5fd1208eb4d4012ce8a7fa16b93a2b50ac4eaf1

File: integ-test/src/test/java/org/opensearch/sql/correctness/runner/connection/JDBCConnection.java (hunk 4 above)
Comment (MitchellGale-BitQuill) [reply integ-test/src/test/java/org/opensearch/sql/correctness/runner/connection/JDBCConnection.java]: Yes, good catch, thanks. Fixed in b5fd1208eb4d4012ce8a7fa16b93a2b50ac4eaf1

**Review by forestmvey — APPROVED** [review]

File: docs/user/dql/functions.rst [hunk 5]
```diff
@@ -234,9 +234,67 @@ CEIL
...
+
+CEILING
+----
+
+Description
+>>>>>>>>>>>
+
+Usage: CEILING(a) takes the ceiling of value a.
+
+Note: CEIL and CEILING functions have the same implementation & functionality
```
Comment (GabeFernandez310) [inline docs/user/dql/functions.rst:273]:
```suggestion
Note: `CEIL`_ and `CEILING`_ functions have the same implementation & functionality
```

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by margarit-h — APPROVED** [review]

**Review by GumpacG — APPROVED** [review]

**Review by Yury-Fridlyand — APPROVED** [review]

**Pull request merged**